zap/src/irgen/des.rs

```diff
@@ -172,6 +242,20 @@ impl Des<'_> {
 		self.push_stmt(Stmt::End);
 	}
 
+	fn check_bitfield(&mut self, var: Var, bits: BitpackMask) -> Expr {
+		Expr::Call(
+			Var::NameIndex(Var::Name("bit32".into()).into(), "btest".into()).into(),
+			None,
+			vec![Expr::Var(var.into()), Expr::BinaryNum(bits)],
+		)
+	}
+
+	fn readboolean(&mut self) -> Expr {
+		let (bits, var) = self.get_bitpack();
+
+		self.check_bitfield(var, bits)
```

Comment:
Was there a reason you swapped the args? You don't have to change it - it just hurt to read!